nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1294,7 +1311,11 @@ def on_load_checkpoint(self, checkpoint) -> None:
                     key.replace('model.', ''): checkpoint_state_dict.pop(key)
                     for key in list(checkpoint_state_dict.keys())
                 }
-                module.load_state_dict(checkpoint_state_dict, strict=True)
+                missing_keys, unexpected_keys = module.load_state_dict(checkpoint_state_dict, strict=False)
+
+                assert len(unexpected_keys) == 0, 'Unexpected key(s) in state_dict: {}. '.format(
+                    ', '.join('"{}"'.format(k) for k in unexpected_keys)
+                )
```

Comment:
Why do we need this? Doesn't strict=True do the same thing?

Reply:
it's different because this doesn't check `missing_keys`. During checkpoint conversion, all the TE `_extra_state` will be in missing keys and we don't want to error. Unexpected keys, on the other hand, is abnormal behavior so we want them to error.

Reply:
Then do we need this in the model, can it just be in the conversion script?

Reply:
I think checking unexpected keys is a must? Are there any cases where we want to allow unexpected keys? (keys exist in the checkpoint file but not in the actual model)

Reply:
I mean keeping strict = True, and then only in ckpt conversion allowing for missing keys

Reply:
done.